rust/connlib/tunnel/src/lib.rs

```diff
@@ -141,7 +142,8 @@ pub struct Tunnel<CB: Callbacks, TRoleState: RoleState> {
     rate_limiter: Arc<RateLimiter>,
     private_key: StaticSecret,
     public_key: PublicKey,
-    peers_by_ip: RwLock<IpNetworkTable<Arc<Peer<TRoleState::Id>>>>,
+    #[allow(clippy::type_complexity)]
+    peers_by_ip: RwLock<IpNetworkTable<(Arc<Peer<TRoleState::Id>>, Arc<DataChannel>)>>,
```

Comment:
Introduce a new type, I think we tidy this up here and we remove the clippy exception

```rs
ConnectedPeer<T> {
  peer: Arc<Peer<T>>,
  channel: Arc<DataChannel>
}
```

Even if we are going to get rid of it latter I'd rather merge it like this.